sdk/nodejs/cmd/run/index.ts

```diff
@@ -111,8 +111,9 @@ function main(args: string[]): void {
     v8Hooks.isInitializedAsync().then(() => {
         require("./run").run(argv, () => {
             programRunning = true;
+        }).then(() =>  {
+            programRunning = false;
```

Comment:
Just to make sure - is `then` right here, or should it be `finally`?  That is, is there a case where an error can flow out of this `run` call and cause `programRunning` not to get set back to `false`?

Reply:
Import additional info for this (in case anyone comes back and looks at this PR).

A critical part of this PR is that it doesn't change any of our existing exception-trapping or exit-code logic.  The *only* thing it changes is setting this bool at the appropriate time.  And the **only** thing this bool controls is whether or not we *print* to the console the stack for an exception.  

In other words, in terms of error-codes, nothing should change, and thus the work we did in https://github.com/pulumi/pulumi/pull/2337 does not regress.